src/components/pages/main/MainPage.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback, useEffect, useState } from "react";
+import BooksList from "./components/booksList/BooksList";
+import { IBook } from "../../../interfaces";
+import { apiService } from "../../../services/ApiService";
+import SearchBar from "./components/searchBar/SearchBar";
+import { FetchStatusEnum } from "../../../enums/FetchStatusEnum";
+import { Outlet, useSearchParams } from "react-router-dom";
+import "./MainPage.css";
+import useLocalStorage from "../../../utils/useLocalStorage";
+import { ThemeContext } from "../../../ThemeContext";
+
+interface MainPageProps {}
+
+const MainPage: React.FC<MainPageProps> = () => {
+  const [theme, setTheme] = useState("light");
+  const [searchParams, setSearchParams] = useSearchParams();
+  const page = searchParams.get("page");
```

Comment:
```suggestion
  const page = Number(searchParams.get("page")) || 0;
```
и все. тебе не надо его писать в стейт. все равно каждый раз как параметры изменятся useSearchParams вызовет ререндер, и значение будет актульным. просто сразу пихай его куда хочешь